oak_functions/client/rust/src/attestation.rs

```diff
@@ -101,106 +99,76 @@ impl AttestationClient {
+
+            let outgoing_message = handshaker
+                .next_step(&incoming_message.body)
+                .context("Couldn't process handshake message")?;
+            if let Some(outgoing_message) = outgoing_message {
+                channel
+                    .send(AttestedInvokeRequest {
+                        body: outgoing_message,
+                    })
+                    .await
+                    .context("Couldn't send handshake message")?;
+            }
+        }
+        let encryptor = handshaker
+            .get_encryptor()
+            .context("Couldn't get encryptor")?;
         Ok(encryptor)
     }
 
-    /// Sends data encrypted by the [`AttestationClient::encryptor`] to the server and returns
-    /// server responses.
+    /// Sends data encrypted by the [`Encryptor`] to the server and returns server responses.
     /// Returns `Ok(None)` to indicate that the corresponding gRPC stream has ended.
     pub async fn send(
         &mut self,
         request: oak_functions_abi::proto::Request,
     ) -> anyhow::Result<Option<Vec<u8>>> {
-        let encrypted_message = self
+        let encrypted_request = self
             .encryptor
             .encrypt(&request.body)
-            .context("Couldn't encrypt data")?;
-        let data_request = AttestedInvokeRequest {
-            request_type: Some(RequestType::Request(crate::proto::Request {
-                encrypted_payload: Some(encrypted_message),
-            })),
-        };
-
+            .context("Couldn't encrypt request")?;
         self.channel
-            .send(data_request)
+            .send(AttestedInvokeRequest {
+                body: encrypted_request,
+            })
             .await
             .context("Couldn't send encrypted data request")?;
-        let response = self
+
+        let encrypted_response = self
             .channel
             .receive()
             .await
             .context("Couldn't send encrypted data request")?;
-        let data = if let Some(response) = response {
-            let response_type = response
-                .response_type
-                .context("Couldn't read response type")?;
-            let encrypted_payload = if let ResponseType::EncryptedPayload(data) = response_type {
-                Ok(data)
-            } else {
-                Err(anyhow!("Received incorrect message type"))
-            }?;
-            let payload = self
-                .encryptor
-                .decrypt(&encrypted_payload)
-                .context("Couldn't decrypt data")?;
-            Some(payload)
-        } else {
-            None
+        let response = match encrypted_response {
+            Some(encrypted_response) => {
+                let response = self
+                    .encryptor
+                    .decrypt(&encrypted_response.body)
+                    .context("Couldn't decrypt response")?;
+                Some(response)
+            }
+            None => None,
```

Comment:
This looks like it should just be a `map`

Reply:
`map` doesn't allow to use `?` inside it